lib/new-admin/graphql/modules/userManagement.js

```diff
@@ -0,0 +1,228 @@
+const bcrypt = require('bcrypt')
+const _ = require('lodash/fp')
+const otplib = require('otplib')
+
+const credentials = require('../../../credentials')
+const loginHelper = require('../../login')
+const users = require('../../../users')
+const sessionManager = require('../../../session-manager')
+const T = require('../../../time')
+
+const REMEMBER_ME_AGE = 90 * T.day
+
+function authenticateUser (username, password) {
+  return loginHelper.checkUser(username).then(hashedPassword => {
+    if (!hashedPassword) return null
+    return Promise.all([bcrypt.compare(password, hashedPassword), hashedPassword])
+  }).then(([isMatch, hashedPassword]) => {
+    if (!isMatch) return null
+    return loginHelper.validateUser(username, hashedPassword)
+  }).then(user => {
+    if (!user) return null
+    return user
+  }).catch(e => {
+    console.error(e)
+  })
+}
+
+const get2FASecret = (username, password) => {
+  return authenticateUser(username, password).then(user => {
+    if (!user) return null
+
+    const secret = otplib.authenticator.generateSecret()
+    const otpauth = otplib.authenticator.keyuri(username, 'Lamassu Industries', secret)
+    return { secret, otpauth }
+  })
+}
+
+const confirm2FA = (codeArg, context) => {
+  const code = codeArg
+  const requestingUser = context.req.session.user
+
+  if (!requestingUser) return false
+
+  return users.get2FASecret(requestingUser.id).then(user => {
+    const secret = user.twofa_code
+    const isCodeValid = otplib.authenticator.verify({ token: code, secret: secret })
+
+    if (!isCodeValid) return false
+    return true
```

Comment:
if isCodeValid is a boolean we can just do return isCodeValid otherwise we can also do return !!isCodeValid